File: IEEE488Interface.py
class IEEE488:

	def __init__(self, ref):
		self.this=ref
	
	def setClearStatus(self):
		self.this.sc.enviar_comando("*CLS\n")
		
	def setBitsStandarEventRegister(self, value_):
		self.this.sc.enviar_comando("*CLS "+ str(value_) +"\n")
		
		#Querys
	
	def getStandardEventEnableRegister(self):
		self.this.sc.enviar_comando("*ESE?\n")
		return self.this.sc.obtener_respuesta()
		
	def getStandardEventRegister_Error(self):
		self.this.sc.enviar_comando("*ESR?\n")
		return self.this.sc.obtener_respuesta()

	def getMeterIdentificationString(self):
		self.this.sc.enviar_comando("*IDN?\n")
		return self.this.sc.obtener_respuesta()
	
	def getOperationCompleteStatus(self):
		self.this.sc.enviar_comando("*OPC?\n")

	def setOperationCompleteBit(self):
		self.this.sc.enviar_comando("*OPC\n")

	def setPowerOnStatusClear(self, Status_):
		self.this.sc.enviar_comando("*PSC "+str(Status_)+"\n")
		
	def getsPowerOnStatusClearSetting(self):
		self.this.sc.enviar_comando("*PSC?\n")

	def setResetMeter(self):
		self.this.sc.enviar_comando("*RST\n")
		
	def setBitsStatusByteRegister(self, value_):
		self.this.sc.enviar_comando("*SRE "+ str(value_) +"\n")

	def getBitsStatusByteRegister(self):
		self.this.sc.enviar_comando("*SRE?\n")
		return self.this.sc.obtener_respuesta()
		
	def getStatusByteSummaryRegister(self):
		self.this.sc.enviar_comando("*STB?\n")
		return self.this.sc.obtener_respuesta()
	
	def getSelfTestStatus(self):
		self.this.sc.enviar_comando("*TST?\n")
		return self.this.sc.obtener_respuesta()

File: DriverDMM4050/IEEE488Interface.c
#include "IEEE488Interface.h"

status_ setClearStatus()
{
    char query1[10];
    snprintf(query1,sizeof(query1),"*CLS?\n");
    return Configuration(query1);
}

//Querys

status_ getStandardEventEnableRegister_()
{
    char query1[10];
    snprintf(query1,sizeof(query1),"*ESE?\n");
    return Query(query1);
}

status_ getStandardEventRegister_Error()
{
    char query1[10];
    snprintf(query1,sizeof(query1),"*ESR?\n");
    return Query(query1);
}

status_ getOperationCompleteStatus()
{
    char query1[10];
    snprintf(query1,sizeof(query1),"*OPC?\n");
    return Query(query1);
}

status_ setOperationCompleteBit()
{
    char query1[10];
    snprintf(query1,sizeof(query1),"*OPC\n");
    return Configuration(query1);
}

status_ setPowerOnStatusClear(int bit_)
{
    char query1[30];
    snprintf(query1,sizeof(query1),"*PSC %d\n",bit_);
    return Configuration(query1);
}

status_ getPowerOnStatusClearSetting()
{
    char query1[10];
    snprintf(query1,sizeof(query1),"*PSC?\n");
    return Query(query1);
}

status_ setBitsStatusByteRegister_(int bit_)
{
    char query1[30];
    snprintf(query1,sizeof(query1),"*SRE %d\n",bit_);
    return Configuration(query1);
}

status_ getBitsStatusByteRegister_()
{
    char query1[10];
    snprintf(query1,sizeof(query1),"*SRE?\n");
    return Query(query1);
}

status_ getStatusByteSummaryRegister_()
{
    char query1[10];
    snprintf(query1,sizeof(query1),"*STB?\n");
    return Query(query1);
}

status_ getSelfTestStatus()
{
    char query1[10];
    snprintf(query1,sizeof(query1),"*TST?\n");
    return Query(query1);
}

